.circleci/config.yml

```diff
@@ -124,6 +127,29 @@ jobs:
           paths:
             - "*"
 
+  build_x86_linux_cxx17:
+    docker:
+      - image: buildpack-deps:disco
+    environment:
+      TERM: xterm
+      COVERAGE: "ON"
+      CMAKE_OPTIONS: -DCMAKE_TOOLCHAIN_FILE=cmake/toolchains/cxx17.cmake
+    steps:
+      - checkout
+      - run:
+          name: Install build dependencies
+          command: |
+            apt-get -qq update
+            apt-get -qy install cmake libboost-regex-dev libboost-filesystem-dev libboost-test-dev libboost-system-dev libboost-program-options-dev libcvc4-dev
+            ./scripts/install_obsolete_jsoncpp_1_7_4.sh
+      - run: *setup_prerelease_commit_hash
+      - run: *run_build
+      - store_artifacts: *solc_artifact
+      - persist_to_workspace:
+          root: build
+          paths:
+            - "*"
```

Comment:
Why is this needed?

Reply:
The block was copied from `build_x86_linux`. Even `build_x86_archlinux` saves some artifacts. Do we really need to save artifacts for these targets not used for testing?

Reply:
For archlinux there's ``test_x86_archlinux`` running the non-ipc tests, which needs some artifacts. And the artifacts of ``build_x86_linux`` are used in ``test_x86_linux``, but I guess we don't need to store any artifacts here, if we don't want to test the C++17 build.

Reply:
Some rules only store the binaries, which is sufficient. Other rules store the whole build directory because that is needed if you want to calculate coverage.